compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1403,12 +1405,16 @@ public BType transform(BLangFiniteTypeNode finiteTypeNode, AnalyzerData data) {
                 finiteTypeNode.pos, SOURCE);
 
         BFiniteType finiteType = new BFiniteType(finiteTypeSymbol);
-        for (BLangExpression literal : finiteTypeNode.valueSpace) {
-            literal.setBType(symTable.getTypeFromTag(literal.getBType().tag));
-            finiteType.addValue(literal);
+
+        // In case finiteTypeNode has unary expressions in the value space, we need to
+        // convert them into numeric literals first.
+        semanticAnalyzer.analyzeNode(finiteTypeNode, data.env);
```

Comment:
We shouldn't use the semantic analyzer here. Why can't this be done at semantic analysis.

Reply:
Did we create an issue for this?